diff --git a/config/fs-router.js b/config/fs-router.js
index 994c49a1ce3e4399c12c729300ca278adc6e2d36..a201bf418a180699cb965dcfca3c898a90afc321 100644
--- a/config/fs-router.js
+++ b/config/fs-router.js
@@ -31,7 +31,6 @@ export class SolidStartClientFileRouter extends BaseFileSystemRouter {
         },
         $$route: undefined,
         path,
-        filePath: src
       };
     }
 
@@ -47,12 +46,11 @@ export class SolidStartClientFileRouter extends BaseFileSystemRouter {
         },
         $$route: hasRouteConfig
           ? {
-              src: src,
-              pick: ["route"]
-            }
+            src: src,
+            pick: ["route"]
+          }
           : undefined,
         path,
-        filePath: src
       };
     }
   }
@@ -102,7 +100,6 @@ export class SolidStartServerFileRouter extends BaseFileSystemRouter {
         },
         $$route: undefined,
         path,
-        filePath: src
       };
     }
 
@@ -116,19 +113,18 @@ export class SolidStartServerFileRouter extends BaseFileSystemRouter {
         $component:
           !this.config.dataOnly && hasDefault
             ? {
-                src: src,
-                pick: ["default", "$css"]
-              }
+              src: src,
+              pick: ["default", "$css"]
+            }
             : undefined,
         $$route: hasRouteConfig
           ? {
-              src: src,
-              pick: ["route"]
-            }
+            src: src,
+            pick: ["route"]
+          }
           : undefined,
         ...createHTTPHandlers(src, exports),
         path,
-        filePath: src
       };
     }
   }
diff --git a/dist/router/lazyRoute.js b/dist/router/lazyRoute.js
index 463636a16572208dfbd801e758dc45c0657ddccc..a0a2daa85fb136e1aafb273c350ba2e881678fff 100644
--- a/dist/router/lazyRoute.js
+++ b/dist/router/lazyRoute.js
@@ -39,8 +39,9 @@ export default function lazyRoute(component, clientManifest, serverManifest, exp
         else {
             const mod = await component.import();
             const Component = mod[exported];
-            let assets = await clientManifest.inputs?.[component.src].assets();
-            const styles = assets.filter((asset) => asset.tag === "style" || asset.attrs.rel === "stylesheet");
+            // let assets = await clientManifest.inputs?.[component.src].assets();
+            // const styles = assets.filter((asset) => asset.tag === "style" || asset.attrs.rel === "stylesheet");
+            const styles = []
             if (typeof window !== "undefined") {
                 preloadStyles(styles);
             }
diff --git a/dist/server/StartServer.jsx b/dist/server/StartServer.jsx
index 0c6a9fe3121a88520db252570a8b5997a36e0452..b1d8c9dc951c91739126bf2524d075e3038f531e 100644
--- a/dist/server/StartServer.jsx
+++ b/dist/server/StartServer.jsx
@@ -52,35 +52,35 @@ export function StartServer(props) {
             else if (import.meta.env.DEV)
                 console.warn("No route matched for preloading js assets");
         }
-        assets = await Promise.all(assetPromises).then(a => 
-        // dedupe assets
-        [...new Map(a.flat().map(item => [item.attrs.key, item])).values()].filter(asset => import.meta.env.START_ISLANDS
-            ? false
-            : asset.attrs.rel === "modulepreload" &&
+        assets = await Promise.all(assetPromises).then(a =>
+            // dedupe assets
+            [...new Map(a.flat().map(item => [item.attrs.key, item])).values()].filter(asset => import.meta.env.START_ISLANDS
+                ? false
+                : asset.attrs.rel === "modulepreload" &&
                 !context.assets.find((a) => a.attrs.key === asset.attrs.key)));
     });
     useAssets(() => (assets.length ? assets.map(m => renderAsset(m)) : undefined));
     return (<NoHydration>
-      {docType}
-      <TopErrorBoundary>
-        <props.document assets={<>
-              <HydrationScript />
-              {context.assets.map((m) => renderAsset(m, nonce))}
+        {docType}
+        <TopErrorBoundary>
+            <props.document assets={<>
+                <HydrationScript />
+                {context.assets.map((m) => renderAsset(m, nonce))}
             </>} scripts={nonce ? (<>
-                <script nonce={nonce} innerHTML={`window.manifest = ${JSON.stringify(context.manifest)}`}/>
-                <script type="module" nonce={nonce} async src={import.meta.env.MANIFEST["client"].inputs[import.meta.env.MANIFEST["client"].handler].output.path}/>
-              </>) : (<>
-                <script innerHTML={`window.manifest = ${JSON.stringify(context.manifest)}`}/>
-                <script type="module" async src={import.meta.env.MANIFEST["client"].inputs[import.meta.env.MANIFEST["client"].handler].output.path}/>
-              </>)}>
-          {!import.meta.env.START_ISLANDS ? (<Hydration>
-              <ErrorBoundary>
-                <App />
-              </ErrorBoundary>
-            </Hydration>) : (<ErrorBoundary>
-              <App />
-            </ErrorBoundary>)}
-        </props.document>
-      </TopErrorBoundary>
+                <script nonce={nonce} innerHTML={`window.manifest = ${JSON.stringify(context.manifest)}`} />
+                <script type="module" nonce={nonce} async src={import.meta.env.MANIFEST["client"].inputs[import.meta.env.MANIFEST["client"].handler].output.path} />
+            </>) : (<>
+                <script innerHTML={`window.manifest = {}`} />
+                <script type="module" async src={import.meta.env.MANIFEST["client"].inputs[import.meta.env.MANIFEST["client"].handler].output.path} />
+            </>)}>
+                {!import.meta.env.START_ISLANDS ? (<Hydration>
+                    <ErrorBoundary>
+                        <App />
+                    </ErrorBoundary>
+                </Hydration>) : (<ErrorBoundary>
+                    <App />
+                </ErrorBoundary>)}
+            </props.document>
+        </TopErrorBoundary>
     </NoHydration>);
 }
