diff --git a/dist/routing.js b/dist/routing.js
index 0a33e859ba30d467e5fc098f562c7119513dfb43..ad0c3e682dc32225c1b07292fc8ae3c9ae900eb5 100644
--- a/dist/routing.js
+++ b/dist/routing.js
@@ -1,8 +1,8 @@
-import { createComponent, createContext, createMemo, createRenderEffect, createSignal, on, onCleanup, untrack, useContext, startTransition, resetErrorBoundaries } from "solid-js";
-import { isServer, delegateEvents } from "solid-js/web";
+import { createComponent, createContext, createMemo, createRenderEffect, createSignal, on, onCleanup, resetErrorBoundaries, startTransition, untrack, useContext } from "solid-js";
+import { delegateEvents, isServer } from "solid-js/web";
 import { normalizeIntegration } from "./integration";
 import { createBeforeLeave } from "./lifecycle";
-import { createMemoObject, extractSearchParams, invariant, resolvePath, createMatcher, joinPaths, scoreRoute, mergeSearchString, expandOptionals } from "./utils";
+import { createMatcher, createMemoObject, expandOptionals, extractSearchParams, invariant, joinPaths, mergeSearchString, resolvePath, scoreRoute } from "./utils";
 const MAX_REDIRECTS = 100;
 export const RouterContextObj = createContext();
 export const RouteContextObj = createContext();
@@ -187,7 +187,7 @@ export function createRouterContext(integration, base = "", data, out) {
     const parsePath = utils.parsePath || (p => p);
     const renderPath = utils.renderPath || (p => p);
     const beforeLeave = utils.beforeLeave || createBeforeLeave();
-    const basePath = resolvePath("", base);
+    const basePath = resolvePath("", base) || "/";
     const output = isServer && out
         ? Object.assign(out, {
             matches: [],