diff --git a/config/fs-router.js b/config/fs-router.js
index 6d92c763c90ff1579b706511c7e8781534eb3e5a..dfdfcc19d0f4af1236dfd5796db01428f08626f8 100644
--- a/config/fs-router.js
+++ b/config/fs-router.js
@@ -31,7 +31,6 @@ export class SolidStartClientFileRouter extends BaseFileSystemRouter {
         },
         $$route: undefined,
         path,
-        filePath: src
       };
     }
 
@@ -52,7 +51,6 @@ export class SolidStartClientFileRouter extends BaseFileSystemRouter {
           }
           : undefined,
         path,
-        filePath: src
       };
     }
   }
@@ -109,7 +107,6 @@ export class SolidStartServerFileRouter extends BaseFileSystemRouter {
         },
         $$route: undefined,
         path,
-        filePath: src
       };
     }
 
@@ -135,7 +132,6 @@ export class SolidStartServerFileRouter extends BaseFileSystemRouter {
           : undefined,
         ...createHTTPHandlers(src, exports),
         path,
-        filePath: src
       };
     }
   }
diff --git a/dist/router/lazyRoute.js b/dist/router/lazyRoute.js
index 463636a16572208dfbd801e758dc45c0657ddccc..e713d0c78f1a77257a0421601f065d5652fc6246 100644
--- a/dist/router/lazyRoute.js
+++ b/dist/router/lazyRoute.js
@@ -39,8 +39,7 @@ export default function lazyRoute(component, clientManifest, serverManifest, exp
         else {
             const mod = await component.import();
             const Component = mod[exported];
-            let assets = await clientManifest.inputs?.[component.src].assets();
-            const styles = assets.filter((asset) => asset.tag === "style" || asset.attrs.rel === "stylesheet");
+            const styles = []
             if (typeof window !== "undefined") {
                 preloadStyles(styles);
             }
diff --git a/dist/server/StartServer.jsx b/dist/server/StartServer.jsx
index 0c6a9fe3121a88520db252570a8b5997a36e0452..b4fc8d14706eced99b022113af8846af75be393b 100644
--- a/dist/server/StartServer.jsx
+++ b/dist/server/StartServer.jsx
@@ -64,13 +64,14 @@ export function StartServer(props) {
       {docType}
       <TopErrorBoundary>
         <props.document assets={<>
-              <HydrationScript />
               {context.assets.map((m) => renderAsset(m, nonce))}
             </>} scripts={nonce ? (<>
+                <HydrationScript />
                 <script nonce={nonce} innerHTML={`window.manifest = ${JSON.stringify(context.manifest)}`}/>
                 <script type="module" nonce={nonce} async src={import.meta.env.MANIFEST["client"].inputs[import.meta.env.MANIFEST["client"].handler].output.path}/>
               </>) : (<>
-                <script innerHTML={`window.manifest = ${JSON.stringify(context.manifest)}`}/>
+                <HydrationScript />
+                <script innerHTML={`window.manifest={}`}/>
                 <script type="module" async src={import.meta.env.MANIFEST["client"].inputs[import.meta.env.MANIFEST["client"].handler].output.path}/>
               </>)}>
           {!import.meta.env.START_ISLANDS ? (<Hydration>
