diff --git a/index.js b/index.js
index 7a593281c2c7baa54337be9a1ec6d658575745a7..a34b0fd2e8900ddb9018854a91a74fbdcab2a9a0 100644
--- a/index.js
+++ b/index.js
@@ -23,7 +23,7 @@ export default function () {
       return `http://localhost:${process.env.PORT}`;
     },
     async build(config, builder) {
-      if(!config?.solidOptions?.ssr) throw new Error('solid-start-static needs ssr to be enabled for pre-rendering routes at build time');
+      if (!config?.solidOptions?.ssr) throw new Error('solid-start-static needs ssr to be enabled for pre-rendering routes at build time');
       const ssrExternal = config?.ssr?.external || [];
       await builder.client(join(config.root, "dist", "public"));
       await builder.server(join(config.root, ".solid", "server"));
@@ -63,7 +63,7 @@ export default function () {
         ...(config.solidOptions.prerenderRoutes || [])
       ];
       renderStatic(
-        routes.map(url => ({
+        routes.map(url => decodeURI(url)).map(url => ({
           entry: join(config.root, "dist", "server.js"),
           output: join(
             pathToDist,