diff --git a/lib/plugins/routes.js b/lib/plugins/routes.js
index 2b08cd1da51fd6675a383759641fa33b7aa0fdf5..c1f490c77cb42e89c499fec2e65d667558c0d8b7 100644
--- a/lib/plugins/routes.js
+++ b/lib/plugins/routes.js
@@ -77,18 +77,8 @@ export function routes() {
 							.join("&")}`;
 						return {
 							src: isBuild ? relative(root, buildId) : buildId,
-							build: isBuild
-								? `_$() => import(/* @vite-ignore */ '${buildId}')$_`
-								: undefined,
 							import:
-								router.target === "server"
-									? `_$() => import(/* @vite-ignore */ '${buildId}')$_`
-									: `_$(() => { const id = '${relative(
-											root,
-											buildId,
-									  )}'; return import(/* @vite-ignore */ import.meta.env.MANIFEST['${
-											router.name
-									  }'].inputs[id].output.path) })$_`,
+								`_$() => import(/* @vite-ignore */ '${buildId}')$_`
 						};
 					}
 					return v;
@@ -146,8 +136,8 @@ function jsCode() {
 
 		return Object.keys(id).length > 0
 			? `{ ${Object.keys(id)
-					.map((k) => `${k} as ${id[k]}`)
-					.join(", ")} }`
+				.map((k) => `${k} as ${id[k]}`)
+				.join(", ")} }`
 			: "";
 	};
 
@@ -155,12 +145,10 @@ function jsCode() {
 		return `${[...imports.keys()]
 			.map(
 				(i) =>
-					`import ${
-						imports.get(i).default
-							? `${imports.get(i).default}${
-									Object.keys(imports.get(i)).length > 1 ? ", " : ""
-							  }`
-							: ""
+					`import ${imports.get(i).default
+						? `${imports.get(i).default}${Object.keys(imports.get(i)).length > 1 ? ", " : ""
+						}`
+						: ""
 					} ${getNamedExport(i)} from '${i}';`,
 			)
 			.join("\n")}`;
